webdriver/tests/support/fixtures.py

```diff
@@ -127,19 +147,35 @@ def session(capabilities, configuration, request):
     deep_update(caps, capabilities)
     caps = {"alwaysMatch": caps}
 
-    # If there is a session with different capabilities active, end it now
+    is_cur_bidi = type(_current_session) == webdriver.BidiSession
+    # If there is a session with different capabilities active or the current session
+    # is of different type than the one we would like to create, end it now.
     if _current_session is not None and (
-            caps != _current_session.requested_capabilities):
-        _current_session.end()
+            (not _current_session.match(caps)) or
```

Comment:
Can the capabilities match but the bidi value mismatch?

Reply:
Yes, user can specify the webSocketUrl capability to true with or without @pytest.mark.bidi(True).
If user specify the webSocketUrl to true without the marker first and then do it again with the same capability but with the marker. The capabilities match but bidi value mismatch. In this case, we still create a new instance of session. Normal use case would be to just use the marker.